Documentation/Concepts/events.md

```diff
@@ -0,0 +1,133 @@
+
+### Naming
+To indicate that the event “has happened in the past”, it should be named as a verb in the past tense.  Often it can contain the name of the entity that the change or action is affecting.
+
+- ✅ `DishPrepared`
+- ✅ `ItemAddedToCart`
+- ❌ `StartCooking`
+- ❌ `AddItemToCart`
+
+## “within your system”
+An event represents something interesting that you wish to capture in your system. Instead of seeing state changes and actions as side effects, they are explicitly modeled within the system and captured within the name, state and shape of our Event.
+
+> State transitions are an important part of our problem space and should be modeled within our domain — Greg Young
+
+### Naming
+An event should be expressed in language that makes sense in the domain, also known as [Ubiquitous Language](https://martinfowler.com/bliki/UbiquitousLanguage.html). You should avoid overly technical/CRUD-like events where such terms are not used in the domain.
+
+For example, in the domain of opening up the kitchen for the day and adding a new item to the menu:
+- ✅ `KitchenOpened`
+- ✅ `DishAddedToMenu`
+- ❌ `TakeoutServerReady`
+- ❌ `MenuListingElementUpdated`
+
+## Main structure of an Event
+This is a simplified structure of the main parts of an event.
+
+```csharp
+Event {
+    Content object
+    EventSourceId Guid
+    EventType {
+        EventTypeId Guid
+        Generation int
+    }
+    Public bool
+}
+```
+
+For the whole structure of an event as defined in the gRPC, please check [Contracts](https://github.com/dolittle/Contracts/tree/master/Source/Runtime/Events).
+
+### Content
+This is the object the user commits.
+
+### EventSourceId
+`EventSourceId` represents the source of the event like a "primary key" in a traditional database. By default, [partitioned event handlers]({{< ref "event_handlers_and_filters#event-handlers" >}}) use it as the `PartitionId`.
+
+### EventType
+An `EventType` is a combination of an `EventTypeId` to uniquely identify the type of event it is and the event types `Generation`.
+This decouples the event from a programming language and enables the renaming of events as the domain language evolves.
+Simply put, event types are a wrapper for the actual type of your event. They mask the underlying language-specific type implementation and conventions.
+
+The Runtime doesn't know or care about the event's content, properties, or type (in its respective programming language). The Runtime saves the event to the event store and then calls the respective [EventHandlers & Filters]({{< ref "event_handlers_and_filters" >}}). For this event to be serialized to JSON and then deserialized back to a type that the client's filters and event handlers understand, an event type is required.
```

Comment:
I would formalise the "event store part" a bit differently.
```suggestion
The Runtime doesn't know or care about the event's content, properties, or type (in its respective programming language). The Runtime saves the event to the event log and from that point the event is ready to be processed. You can read more about how events are processed in the section about [EventHandlers & Filters]({{< ref "event_handlers_and_filters" >}}). For this event to be serialized to JSON and then deserialized back to a type that the client's filters and event handlers understand, an event type is required.
```

Reply:
Fixed it and added just a link for it.